keras_cv/models/densenet.py

```diff
@@ -0,0 +1,296 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""DenseNet models for KerasCV.
+
+Reference:
+  - [Densely Connected Convolutional Networks](https://arxiv.org/abs/1608.06993)
+"""
+
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras import backend
+from tensorflow.keras import layers
+
+BN_AXIS = 3
+
+
+def dense_block(x, blocks, name):
```

Comment:
A general style consideration we need to answer for all future models is this:

Should we do:

```python
def dense_block(x, blocks, name=None):
    ...
    return x
```

or:

```python
def DenseBlock(blocks, name=None):
    def apply(x):
        ...
        return x
    return apply
```

The latter enables a more consistent API style:

```python
...
x = layers.Dense(3, activation="relu")(x)
x = layers.DenseBlock(2)(x)
...
```

Reply:
Updated to the more consistent API design